aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp

```diff
@@ -291,10 +389,39 @@ Tensor embedding_bag_4bit_rowwise_offsets(
   return output;
 }
 
-TORCH_LIBRARY_IMPL(quantized, CPU, m) {
-    m.impl("embedding_bag_byte_rowwise_offsets", embedding_bag_byte_rowwise_offsets);
-    m.impl("embedding_bag_4bit_rowwise_offsets", embedding_bag_4bit_rowwise_offsets);
-}
+template <int bit_rate>
+class QEmbeddingBag final {
+ public:
+  static at::Tensor run(
+      const c10::intrusive_ptr<EmbeddingPackedParamsBase>& packed_weight,
+      const Tensor& indices,
+      const Tensor& offsets,
+      const bool /* scale_grad_by_freq */,
+      const int64_t /* mode */,
+      bool sparse,
+      const c10::optional<Tensor>& per_sample_weights_,
+      const c10::optional<Tensor>& compressed_indices_mapping,
+      bool include_last_offset) {
```

Comment:
how do we decide whether these arguments should be in `run` or `prepack`?

Reply:
asking because conv have some arguments(strde, padding etc.) included in prepack API(https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/conv.py#L201) instead of run, I recall it's because we need these parameters when packing the weights. I guess it's clearer to keep them in `run` in this case.

Reply:
Yeah, in this case all these arguments aren't needed for packing, so might get confusing if we use them in prepack.